### ISSUE TITLE
Question about the predictions made by M

### ISSUE THREAD

**Issue by kaiolae** [worldmodels/worldmodels.github.io#14]
Hi @hardmaru , thanks again for the great work in making this paper accessible and reproducible!

I have a question about the predictions made by M:
In the dreams, it seems M is able to produce quite plausible future scenarios. Would you expect that these dreams predict well what actually happens in a given scenario? In other words, if you condition M on a sequence of real images, would you expect first few images of the "dream" to be close to what actually comes next? Did you do any experiments in this direction?
If I understood correctly, one motivation for using an MDN-RNN is that it gives M some creativity and ability to make several different types of predictions. However, does that also limit its ability to predict accurately what _does_ happen in the future?

Best,
-Kai

**Comment by worldmodels** [comment]
Hi @kaiolae 

You can encode previous ground truth frames and ground truth actions into M, and have it predict future frames based on the real ground truth history, and even sample multiple futures conditional on a common past. The accuracy depends on how well the LSTM models the probability distribution of the training data it is given.

One way to visualize this is to try the sketch-rnn demo, which uses basically the same algorithm as M, but rather than predict future z's, it predicts future pen strokes given a common ground truth past:

![sketch-rnn](https://magenta.tensorflow.org/assets/sketch_rnn_demo/img/multi_sketch_mosquito.gif)

Try it here:

https://magenta.tensorflow.org/assets/sketch_rnn_demo/multi_predict.html
